Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/cart/Cart.kt

```diff
@@ -150,12 +161,117 @@ private fun CartContent(
+                background = { offsetX ->
+                    // Adjust padding right for the main Row, while the delete looks like an Icon the padding must be different from zero
+                    val paddingRight = if (offsetX < -15.dp && offsetX > -200.dp) {
+                        ((offsetX / -2))
+                    } else {
+                        0.dp
+                    }
+                    // Row background is uiFloated until the Remove item text appears, then it is replaced with a red Background
+                    val background = if (offsetX >= -200.dp) {
+                        JetsnackTheme.colors.uiFloated
+                    } else {
+                        JetsnackTheme.colors.error
+                    }
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .requiredHeight(132.dp)
+                            .background(background),
+                        horizontalArrangement = Arrangement.End,
+                        verticalAlignment = Alignment.CenterVertically
+
+                    ) {
+                        // The intended size of the rounded icon
+                        val size = when {
+                            (offsetX < 0.dp && offsetX > -120.dp) -> (offsetX * -1) - 10.dp
+                            (offsetX < -120.dp) -> (offsetX * -1)
+                            else -> 0.dp
+                        }
+                        // To avoid icon to be resized by the graphics layer it is placed in another row. Padding right ensures it looks centered
+                        val paddingIcon = if (size > 0.dp) size / 2 else 0.dp
+                        // Conversion for the intended size to a percentage considering initial size is 10
+                        val heightPercentage = (size.value / 10)
+                        Box {
+                            Row(
+                                modifier = Modifier
+                                    .fillMaxWidth()
+                                    .fillMaxHeight()
+                                    .padding(end = paddingRight),
+                                horizontalArrangement = Arrangement.End,
+                                verticalAlignment = Alignment.CenterVertically
+                            ) {
+                                Surface(
+                                    modifier = Modifier
+                                        .size(10.dp)
+                                        .graphicsLayer(
+                                            scaleX = heightPercentage,
+                                            scaleY = heightPercentage,
+                                            shape = CircleShape,
+                                            clip = true
+                                        ),
+                                    color = JetsnackTheme.colors.error
+
+                                ) {
```

Comment:
nit: The empty content block seems strange. We _could_ switch to a `Box` with a `.background` modifier?

Reply:
The Box component doesn't resize by modifying its graphics layer that is why I used the surface, do you have another suggestion?

Reply:
I think this is not relevant anymore

Reply:
It's hard to understand the evolution since this comment, can it be resolved @angeles-bilbao6 ?

Reply:
Right, I am no longer using a Surface with an empty content block